core/components/_helpers/button-prop.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import Button from '../atoms/button'
+export default (prop, overrides = {}) => {
+  if (!prop) return null
+  if (prop.type === Button) return React.cloneElement(prop, overrides)
+
+  if (typeof prop === 'object') {
+    return (
+      <Button icon={prop.icon} onClick={prop.handler} {...overrides}>
```

Comment:
We should proxy through all of the props (not just icon and onClick)

```jsx
<Button icon={props.icon} onClick={props.handler} {...props} {...overrides}>
```

Reply:
Should we? This way of using the props is being deprecated, maybe using other buttons props can be an incentive to move to the new API.